* Quarto-cli bundles their binary deps near quarto.
This patch makes quarto-cli find all of them in QUARTO_BIN_PATH
* Creates QUARTO_BASE_PATH to replace paths relative to QUARTO_BIN_PATH
* tests/test.ts: some test fail and clober the path for future test 
causing them to fail.
tests/smoke/extensions/extension-render-journals.test.ts:
If await quarto([...]); fails Deno.chdir(wd); doesn't seem to fix the problem

--- a/src/command/render/render-shared.ts
+++ b/src/command/render/render-shared.ts
@@ -194,7 +194,7 @@ export async function rswURL(port: number, path: string) {
 async function rswPortToken(port: number) {
   const result = await execProcess(
     {
-      cmd: ["/usr/lib/rstudio-server/bin/rserver-url", String(port)],
+      cmd: ["@GENTOO_PORTAGE_EPREFIX@/usr/bin/rserver-url", String(port)],
       stdout: "piped",
       stderr: "piped",
     },
--- a/src/core/dart-sass.ts
+++ b/src/core/dart-sass.ts
@@ -15,7 +15,7 @@ import { existsSync } from "fs/mod.ts";
 import { warnOnce } from "./log.ts";
 
 export function dartSassInstallDir() {
-  return toolsPath("dart-sass");
+  return toolsPath("");
 }
 
 export async function dartSassVersion() {
@@ -67,7 +67,7 @@ export async function dartCommand(args: string[]) {
     }
 
     const command = Deno.build.os === "windows" ? "sass.bat" : "sass";
-    return toolsPath(join("dart-sass", command));
+    return toolsPath(command);
   };
   const sass = resolvePath();
 
--- a/src/core/devconfig.ts
+++ b/src/core/devconfig.ts
@@ -69,8 +69,7 @@ export function writeDevConfig(config: DevConfig, binPath: string) {
 }
 
 export function readInstalledDevConfig(): DevConfig | null {
-  const binPath = quartoConfig.binPath();
-  const configPath = join(binPath, "..", "config", kDevConfig);
+  const configPath = join(quartoConfig.basePath(), "package","dist", "config", kDevConfig);
   if (existsSync(configPath)) {
     return JSON.parse(Deno.readTextFileSync(configPath)) as DevConfig;
   } else {
--- a/src/core/quarto.ts
+++ b/src/core/quarto.ts
@@ -27,6 +27,7 @@ export interface QuartoConfig {
 let dotenvConfig: Record<string, string>;
 
 export const quartoConfig = {
+  basePath: () => getenv("QUARTO_BASE_PATH"),
   binPath: () => getenv("QUARTO_BIN_PATH"),
   toolsPath: () => join(getenv("QUARTO_BIN_PATH"), "tools"),
   sharePath: () => getenv("QUARTO_SHARE_PATH"),
@@ -104,7 +105,7 @@ function terminatePreview(reason: string, cleanup?: VoidFunction) {
 }
 
 function quartoSrcDir() {
-  return normalizePath(join(quartoConfig.binPath(), "../../../src"));
+  return normalizePath(join(quartoConfig.basePath(), "src"));
 }
 
 function buildPreviewJs(srcDir: string, cleanup?: VoidFunction) {
--- a/src/core/resources.ts
+++ b/src/core/resources.ts
@@ -67,7 +67,7 @@ export function toolsPath(binary: string): string {
     }
   }
 
-  return join(quartoConfig.toolsPath(), binary);
+  return join(quartoConfig.binPath(), binary);
 }
 
 export function pandocBinaryPath(): string {
--- a/tests/test.ts
+++ b/tests/test.ts
@@ -123,6 +123,12 @@ export function test(test: TestDescriptor) {
   const args: Deno.TestDefinition = {
     name: testName,
     async fn(context) {
+      const testDir = join(quartoConfig.basePath(),"tests");
+      if(testDir !== Deno.cwd()){
+        console.log("Testdir leakage:"+Deno.cwd());
+        console.log("changing to    :"+testDir);
+        Deno.chdir(testDir);
+      }
       await initDenoDom();
       const runTest = !test.context.prereq || await test.context.prereq();
       if (runTest) {
@@ -192,7 +198,7 @@ export function test(test: TestDescriptor) {
             ? fromFileUrl(context.origin)
             : (new URL(context.origin)).pathname;
 
-          const quartoRoot = join(quartoConfig.binPath(), "..", "..", "..");
+          const quartoRoot = quartoConfig.basePath();
           const relPath = relative(
             join(quartoRoot, "tests"),
             absPath,
